From 7846f7f5833860020e32512924e477ccf85a63d4 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Mon, 15 Mar 2021 17:52:00 +0530
Subject: [PATCH] Allow vendor_init to set service.adb.tcp.port

	From 32005f671636a92405172903dc3659a8ced5c0fd Mon Sep 17 00:00:00 2001
	From: Hongguang Chen <hgchen@google.com>
	Date: Tue, 2 Feb 2021 18:10:02 +0530

adbd and apps (SystemUI and CTS test apps) need to read it.

BUG: 162205386
Test: Connect to device which sets service.adb.tcp.port in vendor
          partition through TCP adb.

Change-Id: Ia37dd0dd3239381feb2a4484179a0c7847166b29
Tracked-On: OAM-95330
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 prebuilts/api/30.0/private/adbd.te                     | 1 +
 prebuilts/api/30.0/private/app.te                      | 2 ++
 prebuilts/api/30.0/private/compat/26.0/26.0.ignore.cil | 1 +
 prebuilts/api/30.0/private/compat/27.0/27.0.ignore.cil | 1 +
 prebuilts/api/30.0/private/compat/28.0/28.0.ignore.cil | 1 +
 prebuilts/api/30.0/private/compat/29.0/29.0.ignore.cil | 1 +
 prebuilts/api/30.0/private/property_contexts           | 5 ++++-
 prebuilts/api/30.0/private/vendor_init.te              | 3 +++
 prebuilts/api/30.0/public/property.te                  | 1 +
 private/app.te                                         | 2 ++
 private/compat/26.0/26.0.ignore.cil                    | 1 +
 private/compat/27.0/27.0.ignore.cil                    | 1 +
 private/compat/28.0/28.0.ignore.cil                    | 1 +
 private/compat/29.0/29.0.ignore.cil                    | 1 +
 14 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/prebuilts/api/30.0/private/adbd.te b/prebuilts/api/30.0/private/adbd.te
index be4f0f708..e4498b05b 100644
--- a/prebuilts/api/30.0/private/adbd.te
+++ b/prebuilts/api/30.0/private/adbd.te
@@ -89,6 +89,7 @@ set_prop(adbd, exported_ffs_prop)
 
 # Set service.adb.tls.port, persist.adb.wifi. properties
 set_prop(adbd, adbd_prop)
+set_prop(adbd, adbd_config_prop)
 
 # Access device logging gating property
 get_prop(adbd, device_logging_prop)
diff --git a/prebuilts/api/30.0/private/app.te b/prebuilts/api/30.0/private/app.te
index 9882d8f9b..c8392eedf 100644
--- a/prebuilts/api/30.0/private/app.te
+++ b/prebuilts/api/30.0/private/app.te
@@ -41,3 +41,5 @@ neverallow { appdomain -mediaprovider_app } storage_config_prop:file no_rw_file_
 
 # Allow to read graphics related properties.
 get_prop(appdomain, graphics_config_prop)
+
+get_prop(appdomain, adbd_config_prop)
diff --git a/prebuilts/api/30.0/private/compat/26.0/26.0.ignore.cil b/prebuilts/api/30.0/private/compat/26.0/26.0.ignore.cil
index b395855af..9bf8cf854 100644
--- a/prebuilts/api/30.0/private/compat/26.0/26.0.ignore.cil
+++ b/prebuilts/api/30.0/private/compat/26.0/26.0.ignore.cil
@@ -7,6 +7,7 @@
   ( new_objects
     activity_task_service
     adb_service
+    adbd_config_prop
     adbd_exec
     app_binding_service
     apex_data_file
diff --git a/prebuilts/api/30.0/private/compat/27.0/27.0.ignore.cil b/prebuilts/api/30.0/private/compat/27.0/27.0.ignore.cil
index cb500c9e0..27999b9c5 100644
--- a/prebuilts/api/30.0/private/compat/27.0/27.0.ignore.cil
+++ b/prebuilts/api/30.0/private/compat/27.0/27.0.ignore.cil
@@ -7,6 +7,7 @@
   ( new_objects
     activity_task_service
     adb_service
+    adbd_config_prop
     app_binding_service
     apex_data_file
     apex_metadata_file
diff --git a/prebuilts/api/30.0/private/compat/28.0/28.0.ignore.cil b/prebuilts/api/30.0/private/compat/28.0/28.0.ignore.cil
index d24d12d25..2c06dd5a6 100644
--- a/prebuilts/api/30.0/private/compat/28.0/28.0.ignore.cil
+++ b/prebuilts/api/30.0/private/compat/28.0/28.0.ignore.cil
@@ -7,6 +7,7 @@
   ( new_objects
     activity_task_service
     adb_service
+    adbd_config_prop
     apex_data_file
     apex_metadata_file
     apex_mnt_dir
diff --git a/prebuilts/api/30.0/private/compat/29.0/29.0.ignore.cil b/prebuilts/api/30.0/private/compat/29.0/29.0.ignore.cil
index fdea691ea..aabc7d96f 100644
--- a/prebuilts/api/30.0/private/compat/29.0/29.0.ignore.cil
+++ b/prebuilts/api/30.0/private/compat/29.0/29.0.ignore.cil
@@ -9,6 +9,7 @@
     aidl_lazy_test_server_exec
     aidl_lazy_test_service
     adbd_prop
+    adbd_config_prop
     apex_module_data_file
     apex_permission_data_file
     apex_rollback_data_file
diff --git a/prebuilts/api/30.0/private/property_contexts b/prebuilts/api/30.0/private/property_contexts
index 7908bb107..edf508f53 100644
--- a/prebuilts/api/30.0/private/property_contexts
+++ b/prebuilts/api/30.0/private/property_contexts
@@ -48,7 +48,6 @@ log.tag                 u:object_r:log_tag_prop:s0
 log.tag.WifiHAL         u:object_r:wifi_log_prop:s0
 security.perf_harden    u:object_r:shell_prop:s0
 service.adb.root        u:object_r:shell_prop:s0
-service.adb.tcp.port    u:object_r:shell_prop:s0
 service.adb.tls.port    u:object_r:adbd_prop:s0
 persist.adb.wifi.       u:object_r:adbd_prop:s0
 persist.adb.tls_server.enable  u:object_r:system_adbd_prop:s0
@@ -100,6 +99,10 @@ sys.trace.              u:object_r:system_trace_prop:s0
 # Fastbootd protocol control property
 fastbootd.protocol    u:object_r:fastbootd_protocol_prop:s0 exact enum usb tcp
 
+# adbd protoctl configuration property
+service.adb.transport   u:object_r:adbd_config_prop:s0 exact string
+service.adb.tcp.port    u:object_r:adbd_config_prop:s0 exact int
+
 # Boolean property set by system server upon boot indicating
 # if device is fully owned by organization instead of being
 # a personal device.
diff --git a/prebuilts/api/30.0/private/vendor_init.te b/prebuilts/api/30.0/private/vendor_init.te
index 6a68f1fed..83f001d6a 100644
--- a/prebuilts/api/30.0/private/vendor_init.te
+++ b/prebuilts/api/30.0/private/vendor_init.te
@@ -5,3 +5,6 @@ dontaudit vendor_init sysfs:dir write;
 
 # TODO(b/140259336) We want to remove vendor_init in the long term but allow for now
 allow vendor_init system_data_root_file:dir rw_dir_perms;
+
+# Let vendor_init set service.adb.tcp.port.
+set_prop(vendor_init, adbd_config_prop)
diff --git a/prebuilts/api/30.0/public/property.te b/prebuilts/api/30.0/public/property.te
index 9a93518d6..43b09db8d 100644
--- a/prebuilts/api/30.0/public/property.te
+++ b/prebuilts/api/30.0/public/property.te
@@ -132,6 +132,7 @@ system_vendor_config_prop(vndk_prop)
 system_vendor_config_prop(virtual_ab_prop)
 
 # Properties with no restrictions
+system_public_prop(adbd_config_prop)
 system_public_prop(audio_prop)
 system_public_prop(bluetooth_a2dp_offload_prop)
 system_public_prop(bluetooth_audio_hal_prop)
diff --git a/private/app.te b/private/app.te
index 2ade955a3..7a648dcc4 100644
--- a/private/app.te
+++ b/private/app.te
@@ -57,6 +57,8 @@ get_prop(appdomain, sendbug_config_prop)
 # Allow to read graphics related properties.
 get_prop(appdomain, graphics_config_prop)
 
+get_prop(appdomain, adbd_config_prop)
+
 # Allow to read persist.config.calibration_fac
 get_prop(appdomain, camera_calibration_prop)
 
diff --git a/private/compat/26.0/26.0.ignore.cil b/private/compat/26.0/26.0.ignore.cil
index 1599a3e2b..a2660918c 100644
--- a/private/compat/26.0/26.0.ignore.cil
+++ b/private/compat/26.0/26.0.ignore.cil
@@ -7,6 +7,7 @@
   ( new_objects
     activity_task_service
     adb_service
+    adbd_config_prop
     adbd_exec
     app_binding_service
     apex_data_file
diff --git a/private/compat/27.0/27.0.ignore.cil b/private/compat/27.0/27.0.ignore.cil
index 202baaf48..fc8682889 100644
--- a/private/compat/27.0/27.0.ignore.cil
+++ b/private/compat/27.0/27.0.ignore.cil
@@ -9,6 +9,7 @@
     aaudio_config_prop
     activity_task_service
     adb_service
+    adbd_config_prop
     app_binding_service
     apex_data_file
     apex_metadata_file
diff --git a/private/compat/28.0/28.0.ignore.cil b/private/compat/28.0/28.0.ignore.cil
index d24d12d25..2c06dd5a6 100644
--- a/private/compat/28.0/28.0.ignore.cil
+++ b/private/compat/28.0/28.0.ignore.cil
@@ -7,6 +7,7 @@
   ( new_objects
     activity_task_service
     adb_service
+    adbd_config_prop
     apex_data_file
     apex_metadata_file
     apex_mnt_dir
diff --git a/private/compat/29.0/29.0.ignore.cil b/private/compat/29.0/29.0.ignore.cil
index 10790468f..5f80f732c 100644
--- a/private/compat/29.0/29.0.ignore.cil
+++ b/private/compat/29.0/29.0.ignore.cil
@@ -8,6 +8,7 @@
     aidl_lazy_test_server
     aidl_lazy_test_server_exec
     aidl_lazy_test_service
+    adbd_config_prop
     adbd_prop
     apex_module_data_file
     apex_permission_data_file
-- 
2.17.1

